Remotion/ObjectBinding/Web.ClientScript/Scripts/BocList.ts

```diff
@@ -492,116 +498,125 @@ class BocList
+        var checkStatus = (checkbox as HTMLInputElement).checked
+        document.querySelectorAll('input[name="' + checkName + '"]').forEach(element =>
+        {
+          (element as HTMLInputElement).checked = checkStatus
+        });
+      }) as EventListener)
     });
     // BocList_FixHeaderSize() with timeout needs to be called after setup. This is already taken care of at the call-site.
   }
 
-  private static FixHeaderSize(scrollableContainer: JQuery): void
+  private static FixHeaderSize(scrollableContainer: HTMLElement): void
   {
-    var realTable = scrollableContainer.children('table').first();
-    var realTableWidth = realTable[0].offsetWidth;
-    var previousRealTableWidth = realTable.data("bocListPreviousRealTableWidth");
+    var realTable = scrollableContainer.querySelector<HTMLTableElement>(':scope>table')!;
+    var realTableWidth = BocList.getComputedWidth(realTable);
+    var previousRealTableWidth = parseInt(realTable.dataset.bocListPreviousRealTableWidth!)
     if (previousRealTableWidth == realTableWidth)
       return;
-    realTable.data("bocListPreviousRealTableWidth", realTableWidth);
+    realTable.dataset.bocListPreviousRealTableWidth = realTableWidth.toString()
   
-    var realTableHead = realTable.eq(0).find('thead');
-    var realTableHeadRow = realTableHead.children().eq(0);
-    var realTableHeadRowChildren = realTableHeadRow.children();
+    var realTableHead = realTable.querySelector('thead')!;
+    var realTableHeadRow = realTableHead.children[0];
+    var realTableHeadRowChildren = realTableHeadRow.children;
   
-    var fakeTableHeadContainer = scrollableContainer.parent().children('div.bocListFakeTableHead').first();
-    var fakeTableHead = fakeTableHeadContainer.find('thead');
-    var fakeTableHeadRow = fakeTableHead.children().eq(0);
-    var fakeTableHeadRowChildren = fakeTableHeadRow.children();
+    var fakeTableHeadContainer = scrollableContainer.parentNode!.querySelector<HTMLTableElement>(':scope>div.bocListFakeTableHead')!;
+    var fakeTableHead = fakeTableHeadContainer.querySelector('thead')!;
+    var fakeTableHeadRow = fakeTableHead.children[0];
+    var fakeTableHeadRowChildren = fakeTableHeadRow.children;
   
     // store cell widths in array
     var realTableHeadCellWidths = new Array();
-    realTableHeadRowChildren.each(function (this: HTMLElement, index)
+    Array.from(realTableHeadRowChildren).forEach(function (element: Element, index: number)
     {
-      realTableHeadCellWidths[index] = $(this).width();
+      realTableHeadCellWidths[index] = BocList.getComputedWidth(element as HTMLElement);
     });
   
     // apply widths to fake header
-    fakeTableHeadRowChildren.width(function (index, itemWidth)
+    Array.from(fakeTableHeadRowChildren).forEach((element: Element, index: number) =>
     {
-      var width = realTableHeadCellWidths[index];
-      return width;
+      (element as HTMLElement).style.width = realTableHeadCellWidths[index] + 'px';
     });
```

Comment:
If we need any of the array methods we should convert to an array when declaring the variable.